app/controllers/CreateOrAmendSavingsController.scala

```diff
@@ -35,12 +35,13 @@ class CreateOrAmendSavingsController @Inject()(createOrAmendSavingsService: Crea
 
   def createOrAmendSavings(nino: String, taxYear: Int): Action[AnyContent] = authorisedAction.async { implicit user =>
     user.request.body.asJson.map(_.validate[CreateOrAmendSavingsModel]) match {
-      case Some(JsSuccess(model, _)) => responseHandler(createOrAmendSavingsService.createOrAmendSavings(nino, taxYear, model))
```

Comment:
In this controller can we introduce correlationId change, as the 1903 spec does return 
`400 | INVALID_CORRELATION_ID` Also note the correlationId header name is different for MTD vs downstream, please implement it as per spec.

Reply:
Correlation Id should be passed from the frontend, or are we adding it here too?